src/main/java/de/rabea/Game/Board.java

```diff
@@ -48,23 +82,118 @@ private int emptyCellsCount() {
         return count;
     }
 
-    public Cell switchMark(Cell mark) {
-        if (mark == Cell.X) {
-            return Cell.O;
-        } else {
-            return Cell.X;
+    public List<Integer> indexOfLastCellPerRow() {
+        int length = getDimension();
+        List<Integer> lastCellsPerRow = new ArrayList<Integer>();
+        for (int i = 1; i <= length; i ++) {
+            lastCellsPerRow.add(length * i - 1);
         }
+        return lastCellsPerRow;
     }
 
-    public List<Integer> emptyCells() {
-        List<Integer> emptyCells = new ArrayList<Integer>();
-        int i = 0;
-        for (Cell cell : cells) {
-            if (cell == EMPTY) {
-               emptyCells.add(i);
+    public int getDimension() {
```

Comment:
Consider passing dimension to the constructor (haven't looked but guessing you pass in the number of cells). Much easier to calculate number of cells at board creation time, with the benefit of being able to just return the value here.